examples/point-of-sale/src/client/components/contexts/PaymentProvider.tsx

```diff
@@ -26,7 +27,16 @@ export const PaymentProvider: FC<PaymentProviderProps> = ({ children }) => {
     const { link, recipient, splToken, label, message, requiredConfirmations, connectWallet } = useConfig();
     const { publicKey, sendTransaction } = useWallet();
 
-    const [amount, setAmount] = useState<BigNumber>();
+    const router = useRouter();
+
+    const [amount, setAmount] = useState<BigNumber | undefined>(() => {
+        // Use the amount query param as initial value if set
+        const { amount: queryAmount } = router.query;
+        if (!queryAmount) return undefined;
+        if (Array.isArray(queryAmount)) return new BigNumber(queryAmount[0]);
+        return new BigNumber(queryAmount)
```

Comment:
Hmm, I think we might need to validate the amount in some way. If it doesn't pass a simple regex, we probably don't want to instantiate a `BigNumber` from it.

Reply:
I've added some validation (not NaN, finite, >0) using BigNumber functions. I think that's a bit more straightforward than adding our own regex, and since the amount is displayed on screen + in the wallet it's ok to allow whatever parsing flexibility BigNumber has here. Can add a regex instead if you'd prefer though, I don't have a strong view either way really.

Reply:
Yeah, I think either is fine. My idea was simply to check it against `^\d+(\.\d+)?$` since this is (more or less) what describes a valid manual amount entry into the UI. Zero amounts are _technically_ probably okay to allow (they are allowed in the protocol) but I don't care enough to change this.